CCv0: refactoring common code into tests_common.sh

wainersm opened the pull request:
This is a follow up of https://github.com/kata-containers/tests/pull/4677 where @stevenhorsman suggested to put common test code into a library. While here I changed the name of the `$start_date` variable to a more meaningful one.

cmaf commented:
/test

wainersm commented:
Updated this PR to address a conflict since the merge of commit d5ce71d7402c87ad4264f7e9076109bbff41c137 . I changed the `setup_common()` and `teardown_common()` functions to include the handling of https proxy and dns, which are required for the Intel TDX CI (running behind a firewall).

@arronwy could you please check that this PR fix the regressions introduced on b306e7a726d6f60322236ae34faf38f6f6228177 ?

wainersm commented:
/test

wainersm commented:
/test-fc

arronwy commented:
> Updated this PR to address a conflict since the merge of commit [d5ce71d](https://github.com/kata-containers/tests/commit/d5ce71d7402c87ad4264f7e9076109bbff41c137) . I changed the `setup_common()` and `teardown_common()` functions to include the handling of https proxy and dns, which are required for the Intel TDX CI (running behind a firewall).
> 
> @arronwy could you please check that this PR fix the regressions introduced on [b306e7a](https://github.com/kata-containers/tests/commit/b306e7a726d6f60322236ae34faf38f6f6228177) ?

Thanks @wainersm , I verified this PR works for testcase 1, but still failed for testcase 6 aget api test. I found the root cause is this test case use the agent config file, with this option Kata agent will not parse other config param from command line.

Then, we may aslo need add the `https_proxy` in the agent config file in common setup function, this config need add above `[endpoints]`:
```
sed -i -e '/^\[endpoints\]/i agent.https_proxy = "'${HTTPS_PROXY}'"' agent-configuration-no-exec.toml
```

wainersm commented:
> > Updated this PR to address a conflict since the merge of commit [d5ce71d](https://github.com/kata-containers/tests/commit/d5ce71d7402c87ad4264f7e9076109bbff41c137) . I changed the `setup_common()` and `teardown_common()` functions to include the handling of https proxy and dns, which are required for the Intel TDX CI (running behind a firewall).
> > @arronwy could you please check that this PR fix the regressions introduced on [b306e7a](https://github.com/kata-containers/tests/commit/b306e7a726d6f60322236ae34faf38f6f6228177) ?
> 
> Thanks @wainersm , I verified this PR works for testcase 1, but still failed for testcase 6 aget api test. I found the root cause is this test case use the agent config file, with this option Kata agent will not parse other config param from command line.
> 
> Then, we may aslo need add the `https_proxy` in the agent config file in common setup function, this config need add above `[endpoints]`:
> 
> ```
> sed -i -e '/^\[endpoints\]/i agent.https_proxy = "'${HTTPS_PROXY}'"' agent-configuration-no-exec.toml
> ```

ah, true. I forgot that. I will adjust that and send a new version. Thanks @arronwy !

wainersm commented:
This was updated with:
 * added a handler for proxy on the agent_api tests
 * changed the `tests_common()` proxy handler so that it does create a temporary pod configuration file to avoid overwrite the original under the fixtures directory
 * saving the kernel parameters earlier on `setup_common()`

wainersm commented:
/test

@arronwy could you please test it on your environment to check the proxy handlers work fine?

arronwy commented on integration/containerd/confidential/agent_api.bats:
```diff
@@ -4,56 +4,53 @@
+	local https_proxy="${HTTPS_PROXY:-${https_proxy:-}}"
+	if [ -n "$https_proxy" ]; then
+		local new_file="${BATS_FILE_TMPDIR}/$(basename ${agent_config_file})"
+		echo "https_proxy = '${HTTPS_PROXY}'" > "$new_file"
```
we may need use `https_proxy` instead of `HTTPS_PROXY`

wainersm replied:
Good catch! I just updated this PR with the fix.

arronwy commented:
> /test
> 
> @arronwy could you please test it on your environment to check the proxy handlers work fine?

Thanks @wainersm , Yes, it works well on our environment. Only a minor change needed,.

stevenhorsman commented:
/test

wainersm commented on integration/containerd/confidential/agent_api.bats:
```diff
@@ -4,56 +4,53 @@
+		cat $agent_config_file >> "$new_file"
+		agent_config_file="$new_file"
+		# Print the file content just for the sake of debugging.
+		echo "New agent configure file with https_proxy: $agent_config_file"
```
Good idea. I have a couple of small improvements to those CC tests that I will send in another PR alongside with this suggestion.

wainersm commented:
Updated this PR to address @jodh-intel 's suggestion the https proxy's echo.

wainersm commented:
/test

wainersm commented:
@stevenhorsman this is ready to be merged and it should unlock the TDX CI of @arronwy . However, it once merged I believe it will conflict so badly with Georgina's PR: https://github.com/kata-containers/tests/pull/4729

Is it okay if I merge this anyway?

stevenhorsman commented:
> @stevenhorsman this is ready to be merged and it should unlock the TDX CI of @arronwy . However, it once merged I believe it will conflict so badly with Georgina's PR: #4729
> 
> Is it okay if I merge this anyway?

I'll merge it now as we need to unblock the TDX issues, but then we can work with @George to get that PR working on top of it.
